app/routes/index.tsx

```diff
@@ -24,8 +24,10 @@ import type {Context} from '~/root'
 const empty: Awaited<ReturnType<typeof loadInitialQuestions>> = {data: [], timestamp: ''}
 export const loader = async ({request}: Parameters<LoaderFunction>[0]) => {
   const embed = !!request.url.match(/embed/)
+  const queryFromUrl = !!request.url.match(/[?&]q=/)
   const showInitial = !!request.url.match(/showInitial/)
-  if (embed && !showInitial) return {initialQuestionsData: empty}
+  const hideInitial = (embed || queryFromUrl) && !showInitial
```

Comment:
[NIT] maybe `fetchInitial` and `if(!fetchInitial)`?

Reply:
I'm fine either way => changed